### PR TITLE
[API] revert to double apis

### PR DESCRIPTION
**Pull request by jqueguiner**
@jantic 
c.c. 
@alexandrevicenzi ;-)

I found a better way to not overload the GPU when you want only 1 of the 2 API : use env var when init the docker to avoid un-necessary vRAM usage.

### CODE REVIEW

**Comment by alexandrevicenzi** [comment]
@jqueguiner I believe that API code does not belongs to DeOldify, it should be a different project, but for now is not possible because DeOldify is not a Python package, I still need to finish this.

File: app_utils.py [hunk 1]
```diff
@@ -124,3 +132,64 @@ def get_multi_model_bin(model_list):
     for m in model_list:
         thread.start_new_thread(get_model_bin, m)
 
+
+def unzip(path_to_zip_file, directory_to_extract_to='.'):
```
Comment (alexandrevicenzi) [inline app_utils.py]: Not used?

File: requirements.txt [hunk 2]
```diff
@@ -6,3 +6,4 @@ youtube-dl>=2019.4.17
 jupyterlab
 opencv-python>=3.3.0.10
 pillow
+pyunpack
```
Comment (alexandrevicenzi) [inline requirements.txt]: This dependency is only used by API, requirements.txt is for DeOldify. This dependency is not required to run notebooks or Colab, I believe it should be moved to requirements-api.txt. Perhaps there are more dependencies that should be moved.

File: app_utils.py [hunk 3]
```diff
@@ -124,3 +132,64 @@ def get_multi_model_bin(model_list):
...
+    print("deflating model :" + path_to_zip_file)
+    with zipfile.ZipFile(path_to_zip_file, 'r') as zip_ref:
+        zip_ref.extractall(directory_to_extract_to)
+
+
+def unrar(path_to_rar_file, directory_to_extract_to='.'):
```
Comment (alexandrevicenzi) [inline app_utils.py]: Not used?

File: app.py [hunk 4]
```diff
@@ -34,18 +34,53 @@
 app = Flask(__name__)
 
 
+# define a predict function as an endpoint
+@app.route("/process_video", methods=["POST"])
```
Comment (alexandrevicenzi) [inline app.py]: Being REST like I suggest:

/process/video
/process/image

File: app.py [hunk 5]
```diff
@@ -34,18 +34,53 @@
...
+        callback = send_file(output_path, mimetype='application/octet-stream')
+
+        return callback, 200
+
+    except:
+        traceback.print_exc()
+        return {'message': 'input error'}, 400
+
+    finally:
+        pass
```
Comment (alexandrevicenzi) [inline app.py]: pass is wrong here, it should be removed

File: app.py [hunk 6]
```diff
@@ -71,27 +106,48 @@ def process_image():
...
+    # VIDEO for Video API Only
+    # the purpose is to avoid
+
+    api_type = os.getenv('COLORIZER_API_TYPE', 'BOTH')
+
+    image_colorizer = None
+    video_colorizer = None
+    
+    if api_type == 'IMAGE' or api_type == 'BOTH':
+        artistic_model_url = 'http://storage.gra5.cloud.ovh.net/v1/AUTH_18b62333a540498882ff446ab602528b/pretrained-models/image/deoldify/ColorizeArtistic_gen.pth'
```
Comment (alexandrevicenzi) [inline app.py]: What about DeOldify original models? I do believe that if you want to use OVH models you definitely should remove all this code from DeOldify, this API code does seems part of DeOldify.

Comment (jqueguiner) [reply app.py]: This is not an ovh model it's the same as the dropbox one except that dropbox download is really slow compared the object storage . I can revert it to be dropbox.

Comment (alexandrevicenzi) [reply app.py]: Well, Dropbox can be slow, but you download only once right?

Also, depending on where is your storage you pay traffic usage for all downloads that you will get from this URL, are you sure you want that?

**Comment by jqueguiner** [comment]
> @jqueguiner I believe that API code does not belongs to DeOldify, it should be a different project, but for now is not possible because DeOldify is not a Python package, I still need to finish this.

ok got it I'll let you create a PR to remove the API from the main project.

**Pull request closed without merging**